docs/guide/docker.md

```diff
@@ -0,0 +1,114 @@
+---
+id: docker
+title: 🐳 Build docker image
+description: Build a simple docker image that can be used in production.
+sidebar_position: 8
+---
+
+After we have completed the development of our application, the moment comes when we want to upload it to the real world, most often we use Docker for this, wrapping our application in a container.
+
+To do this we need to follow a few simple steps:
+
+## Step 1 - Setup project
+
+First create folder for our project run the following command in the console:
+
+```bash title="~/"
+mkdir example && cd example
+```
+
+Then we have to init go project:
+```bash title="~/example"
+go mod init example.org/demo
+```
+
+Add **Fiber** to our project:
+```bash title="~/example"
+go get github.com/gofiber/fiber/v2
+```
+
+## Step 2 - Hello world
+
+To start we shold create `main.go` file:
+```bash title="~/example"
+touch main.go
+```
+
+Next we should add following content to our file:
+
+```go title="~/app/main.go"
+package main
+
+import "github.com/gofiber/fiber/v2"
+
+func main() {
+	app := fiber.New()
+
+	app.Get("/", func(c *fiber.Ctx) error {
+		return c.SendString("Hello, World!")
+	})
+
+	app.Listen(":3000")
+}
+```
+
+## Step 3 - create Dockerfile
+
+Run the following command in the console:
+
+```bash
+touch Dockerfile
+```
+
+For the image we will use [Distroless](https://github.com/GoogleContainerTools/distroless) docker image, in the file we should add following content:
+
+```bash title="~/example/Dockerfile"
+FROM golang:1.21 as build
+
+WORKDIR /app
+COPY . .
+
+RUN go mod download && go mod verify
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o /server .
+
+FROM gcr.io/distroless/static-debian12
+COPY --from=build /server .
+
+ENV SOME_VAR=foo
+
+EXPOSE 3000
+CMD ["./server"]
+```
+
+:::info
```

Comment:
I'd say remove everything below this line, except the last step about running it.

Reply:
I agree with you if we are talking about experienced developers (3-5 years of experience) but I tried to cover as large a group of developers as possible.

For example, I wouldn’t read such a section at all because I have a lot of experience with compiled languages and Docker and I can write a config using Docker documentation, but many people who studied with me usually only knew the general principles and how to run containers.

They did not know how to assemble containers and what two-stage assembly is, etc. By adding this information here, I want to close all possible questions. Because otherwise they will ask them here in "Discuss" or in stackoverflow.